log4j-core/src/main/java/org/apache/logging/log4j/core/filter/RegexFilter.java

```diff
@@ -100,59 +242,191 @@ private String targetMessageTest(final Message message) {
                 : message.getFormattedMessage();
     }
 
-    private Result filter(final String msg) {
-        if (msg == null) {
-            return onMismatch;
+    /** {@inheritDoc} */
+    @Override
+    public String toString() {
+        return "useRawMessage=" + useRawMessage + ", pattern=" + pattern.toString();
+    }
+
+    /**
+     * Creates a new builder instance.
+     * @return the new builder instance
+     */
+    @PluginBuilderFactory
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    /**
+     * A {@link RegexFilter} builder instance.
+     */
+    public static final class Builder extends AbstractFilterBuilder<RegexFilter.Builder>
+            implements org.apache.logging.log4j.core.util.Builder<RegexFilter> {
+
+        /* NOTE: LOG4J-3086 - No patternFlags in builder - this functionality has been deprecated/removed. */
+
+        /**
+         * The regular expression to match.
+         */
+        @PluginBuilderAttribute
+        @Required(message = "No 'regex' provided for RegexFilter")
```

Comment:
Please don't provide a failure message for cases where _the failure system_ should be providing the necessary context.

```suggestion
        @Required
```